### ISSUE TITLE
Model not in eval mode while collecting gradients for pruning

### ISSUE THREAD

**Issue by eldarkurtic** [neuralmagic/sparseml#756]
**Describe the bug**
When collecting gradients for pruning with MFACPruningModifier, the model is in the train mode and this has some negative consequences:
1. BatchNorm statistics will be updated while iterating over the samples and this can severely harm model's performance (for example in oneshot pruning of the Mobilenet model, accuracy drops from 55% to 18%)
2. Dropout layers will be turned on, and we usually don't want this type of regularization when estimating curvature of the loss
